jvm/src/main/scala/org/portablescala/reflect/Reflect.scala

```diff
@@ -9,25 +9,65 @@ import java.lang.reflect._
 import org.portablescala.reflect.annotation._
 
 object Reflect {
-  /** Magic to get cross-compiling access to `blackbox.Context` with a fallback
-   *  on `macros.Context`, without deprecation warning in any Scala version.
-   */
-  private object MacroCompat {
-    object Scope1 {
-      object blackbox
+
+  // Macro definitions are enclosed in a dedicated `Macros` object,
+  // so that their metadata (the types involved etc.) don't pollute `Reflect`'s metadata.
+  // This enables using `Reflect`'s methods without `scala-reflect` JAR
+  // https://github.com/scala/bug/issues/8090
+  // https://github.com/xeno-by/sbt-example-paradise210/issues/1#issuecomment-20996354
+  object Macros {
```

Comment:
IIUC the linked issue (in particular the second one; the first one doesn't seem to say anything), even putting this is an `object` within `object Reflect` is likely to cause errors downstream. And if it doesn't, it would be by chance and would be pretty fragile.

IMO we should put this in a separate, top-level objects. Since it unfortunately has to be public, we should put it in `org.portablescala.reflect.internal`.

Reply:
Also, please use `/* */` comments for non-Scaladoc multi-line comments. See https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#non-scaladoc-comments

Reply:
I have actually tested it locally and it worked, but separate object sounds more robust :+1: 
done :heavy_check_mark: